ui/revenuecatui/src/test/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModelTest.kt

```diff
@@ -0,0 +1,260 @@
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.revenuecat.purchases.CustomerInfo
+import com.revenuecat.purchases.Offering
+import com.revenuecat.purchases.Offerings
+import com.revenuecat.purchases.Package
+import com.revenuecat.purchases.PurchaseResult
+import com.revenuecat.purchases.models.StoreTransaction
+import com.revenuecat.purchases.paywalls.PaywallData
+import com.revenuecat.purchases.ui.revenuecatui.PaywallListener
+import com.revenuecat.purchases.ui.revenuecatui.PaywallMode
+import com.revenuecat.purchases.ui.revenuecatui.PaywallOptions
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.TemplateConfiguration
+import com.revenuecat.purchases.ui.revenuecatui.data.testdata.MockApplicationContext
+import com.revenuecat.purchases.ui.revenuecatui.data.testdata.TestData
+import com.revenuecat.purchases.ui.revenuecatui.extensions.getActivity
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.runs
+import io.mockk.verifyOrder
+import junit.framework.TestCase.fail
+import kotlinx.coroutines.delay
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class PaywallViewModelTest {
+    private val defaultOffering = TestData.template2Offering
+
+    private lateinit var purchases: PurchasesType
+    private lateinit var customerInfo: CustomerInfo
+
+    private lateinit var context: Context
+    private lateinit var activity: Activity
+    private lateinit var listener: PaywallListener
+
+    private val offerings = Offerings(
+        defaultOffering,
+        mapOf(
+            TestData.template1Offering.identifier to TestData.template1Offering,
+            TestData.template2Offering.identifier to TestData.template2Offering
+        )
+    )
+
+    @get:Rule
+    var instantExecutorRule = InstantTaskExecutorRule()
+
+    @Before
+    fun setUp() {
+        purchases = mockk()
+        customerInfo = mockk()
+
+        activity = mockk()
+        context = mockk()
+
+        listener = mockk()
+
+        // Allows mocking Context.getActivity
+        mockkStatic("com.revenuecat.purchases.ui.revenuecatui.extensions.ContextExtensionsKt")
+
+        every { context.getActivity() } returns activity
+
+        coEvery { purchases.awaitOfferings() } returns offerings
+        coEvery { purchases.awaitCustomerInfo(any()) } returns customerInfo
+
+        every { listener.onPurchaseStarted(any()) } just runs
+        every { listener.onPurchaseCompleted(any(), any()) } just runs
+        every { listener.onPurchaseError(any()) } just runs
+        every { listener.onRestoreStarted() } just runs
+        every { listener.onRestoreCompleted(any()) } just runs
+        every { listener.onRestoreError(any()) } just runs
+    }
```

Comment:
We should have an `@After` that clears mocks

Reply:
Like this?
```kotlin
@After
internal fun tearDown() {
    clearAllMocks()
}
```